modin/core/dataframe/pandas/partitioning/partition.py

```diff
@@ -366,14 +376,14 @@ def split(self, split_func, num_splits, *args):
         list
             A list of partitions.
         """
-        logger = get_logger()
-        logger.debug(f"ENTER::Partition.split::{self._identity}")
+        log = get_logger()
+        self._is_debug(log) and log.debug(f"ENTER::Partition.split::{self._identity}")
```

Comment:
Is it okay to check for `log.debug(f"ENTER::Partition.split::{self._identity}")` in if statement? What does `log.debug(...)` return?

Reply:
`log.debug(...)` returns `None`.

Reply:
Then I do not understand the use of it in if statement. That looks really confusing. I would prefer to move it under if statement as it is done for Ray here https://github.com/modin-project/modin/pull/5679/files#diff-7a50d6c793451d49f87a91d7d5a444d15c92f8c0de2b45157f5f8b0433001af3R60.

Reply:
one-line condition as it is looks better to me. Logical expressions are executed sequentially, meaning that the computation of the second operand won't be triggered unless the first one is true. This behavior allows writing such fancy one-line checks
```python
...
condition and function_that_should_be_executed_only_if_condition_is_true()
...
```

Reply:
This feature is [declared](https://docs.python.org/3.8/library/stdtypes.html#boolean-operations-and-or-not) by the language. Based on regular lazy evaluation.

In the case of the `if` construct, the amount of code will be doubled. I want to leave it as is.

Reply:
I do not understand why the amount of code will be doubled but ok, let's leave as is. However, before merging align the place I linked above with the current way.

Reply:
I meant the number of lines will double.

Example:
```python
if self._is_debug(log):
    log.debug(f"ENTER::Partition.split::{self._identity}")
```